### PR TITLE
Changing to DB2 documentation updated

### PR DESCRIPTION
**Pull request by Buddhimah**
> https://is.docs.wso2.com/en/5.9.0/administer/changing-to-ibm-db2/ updated for new config model

### CODE REVIEW

File: en/docs/administer/changing-to-ibm-db2.md [hunk 1]
```diff
@@ -1,84 +1,155 @@
...
+      <td><strong>username</strong> and <strong>password</strong></td>
+      <td>The name and password of the database user.</td>
+      </tr>
+      <tr class="even">
+      <td><strong>type</strong></td>
+      <td>The type of the database.</td>
+      </tr>
+      <tr class="even">
+      <td><strong>hostname</strong></td>
+      <td>The hostname of the host where database is hosted.</td>
```
Comment (gomathyk) [inline en/docs/administer/changing-to-ibm-db2.md]: Shall we change this to "The hostname of the hosted database"

File: en/docs/administer/changing-to-ibm-db2.md [hunk 2]
```diff
@@ -1,84 +1,155 @@
...
+                      
+           - `<IS-HOME>/dbscripts/db2.sql`
+           
+   3. If you have a requirement in using workflow feature follow, 
+       [Changing the default database of BPS database](../../administer/changing-datasource-bpsds)
+       
+   4.  Download the DB2 JDBC driver for the version you are using and
+            copy it to the `<IS_HOME>/repository/components/lib` folder  
+    
+    !!! note
```
Comment (gomathyk) [inline en/docs/administer/changing-to-ibm-db2.md]: Add a comma after versions.

File: en/docs/administer/changing-to-ibm-db2.md [hunk 3]
```diff
@@ -1,84 +1,155 @@
...
+        place within the organization.  
+           
+            
 
-Follow the steps below to change the type of the default datasource.
+   2.**Advanced Database Configurations.**
 
-1.  Edit the default datasourceconfigurationin the \<
-    `            IS_HOME>/repository/conf/deployment.toml file as shown below.
+Apart from above basic configurations WSO2 Identity Server supports advanced database configurations.
```
Comment (gomathyk) [inline en/docs/administer/changing-to-ibm-db2.md]: Apart from the basic configurations specified above, WSO2 Identity Server supports some advanced database configurations as well.

File: en/docs/administer/changing-to-ibm-db2.md [hunk 4]
```diff
@@ -90,180 +161,116 @@ Follow the steps below to change the type of the default datasource.
     </tr>
     <tr class="odd">
     <td><p><strong>testOnBorrow</strong></p></td>
-    <td>The indication of whether objects will be validated before being borrowed from the pool. If the object fails to validate, it will be dropped from the pool, and another attempt will be made to borrow another.</td>
+    <td>Whether objects will be validated before being borrowed from the pool. If the object fails to validate, it will be dropped from the pool, and another attempt will be made to borrow another.</td>
```
Comment (gomathyk) [inline en/docs/administer/changing-to-ibm-db2.md]: Indicates whether

File: en/docs/administer/changing-to-ibm-db2.md [hunk 5]
```diff
@@ -90,180 +161,116 @@ Follow the steps below to change the type of the default datasource.
...
     </tr>
     <tr class="even">
-    <td><strong>validationQuery</strong></td>
-    <td>The SQL query that will be used to validate connections from this pool before returning them to the caller.</td>
+    <td><p><strong>defaultAutoCommit</strong></p></td>
+    <td>Whether to commit database changes automatically or not.</td>
```
Comment (gomathyk) [inline en/docs/administer/changing-to-ibm-db2.md]: Indicates whether

File: en/docs/administer/changing-to-ibm-db2.md [hunk 6]
```diff
@@ -90,180 +161,116 @@ Follow the steps below to change the type of the default datasource.
...
+  2.  Add the following JVM option:
+
+       ``` java
+       -Dndatasource.disable.rollbackOnReturn=true \
+       ```
+
+  3.  Navigate to the
+        `               <IS_HOME>/repository/conf/deployment.toml              `
+        file.
+  4.  Disable the `               defaultAutoCommit              `
```
Comment (gomathyk) [inline en/docs/administer/changing-to-ibm-db2.md]: Disable the `               defaultAutoCommit              ` property

File: en/docs/administer/changing-to-ibm-db2.md [hunk 7]
```diff
@@ -90,180 +161,116 @@ Follow the steps below to change the type of the default datasource.
...
+        commitOnReturn="true"
+       ```    
+            
+### Configure the connection pool to rollback pending transactions on connection return
+
+  1.  Navigate to the
+        `<IS_HOME>/repository/conf/deployment.toml`            `
+        file.
+  2.  Disable the
+        `                defaultAutoCommit               ` by
```
Comment (gomathyk) [inline en/docs/administer/changing-to-ibm-db2.md]: Disable the `                defaultAutoCommit               ` property

File: en/docs/administer/changing-to-ibm-db2.md [hunk 8]
```diff
@@ -90,180 +161,116 @@ Follow the steps below to change the type of the default datasource.
...
+ |----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+ | **commitOnReturn**   | If `                defaultAutoCommit               ` =false, then you can set `                commitOnReturn               ` =true, so that the pool can complete the transaction by calling the commit on the connection as it is returned to the pool. However, If `                rollbackOnReturn               ` =true then this attribute is ignored. The default value is false. |
```
Comment (gomathyk) [inline en/docs/administer/changing-to-ibm-db2.md]: However, if the `                rollbackOnReturn               ` property is set to  `true`,

File: en/docs/administer/changing-to-ibm-db2.md (hunk 1 above)
Comment (Buddhimah) [reply en/docs/administer/changing-to-ibm-db2.md]: done

File: en/docs/administer/changing-to-ibm-db2.md (hunk 2 above)
Comment (Buddhimah) [reply en/docs/administer/changing-to-ibm-db2.md]: done

File: en/docs/administer/changing-to-ibm-db2.md (hunk 3 above)
Comment (Buddhimah) [reply en/docs/administer/changing-to-ibm-db2.md]: done

File: en/docs/administer/changing-to-ibm-db2.md (hunk 4 above)
Comment (Buddhimah) [reply en/docs/administer/changing-to-ibm-db2.md]: done

File: en/docs/administer/changing-to-ibm-db2.md (hunk 5 above)
Comment (Buddhimah) [reply en/docs/administer/changing-to-ibm-db2.md]: done

File: en/docs/administer/changing-to-ibm-db2.md (hunk 6 above)
Comment (Buddhimah) [reply en/docs/administer/changing-to-ibm-db2.md]: done

File: en/docs/administer/changing-to-ibm-db2.md (hunk 8 above)
Comment (Buddhimah) [reply en/docs/administer/changing-to-ibm-db2.md]: done

**Pull request merged**